File: src/test/java/com/alibaba/ttl/testmodel/Task.java
package com.alibaba.ttl.testmodel;

import com.alibaba.ttl.TransmittableThreadLocal;
import com.alibaba.ttl.Utils;

import java.util.Map;
import java.util.concurrent.ConcurrentMap;

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
public class Task implements Runnable {
    private final String tag;
    private ConcurrentMap<String, TransmittableThreadLocal<String>> ttlInstances;

    public Task(String tag, ConcurrentMap<String, TransmittableThreadLocal<String>> ttlInstances) {
        this.tag = tag;
        this.ttlInstances = ttlInstances;
    }

    public volatile Map<String, Object> captured;

    @Override
    public void run() {
        try {
            captured = Utils.modifyTtlInstances(tag, ttlInstances);
        } catch (Throwable e) {
            e.printStackTrace();
        }
    }
}

File: src/test/java/com/alibaba/ttl/testmodel/Task.kt
package com.alibaba.ttl.testmodel

import com.alibaba.createChildTtlInstancesAndModifyParentTtlInstances
import com.alibaba.ttl.TransmittableThreadLocal
import java.util.concurrent.ArrayBlockingQueue
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.ConcurrentMap
import java.util.concurrent.TimeUnit

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
class Task(
    private val tag: String,
    private val ttlInstances: ConcurrentMap<String, TransmittableThreadLocal<String>> = ConcurrentHashMap()
) : Runnable {

    private val queue = ArrayBlockingQueue<Map<String, String>>(1);

    val copied: Map<String, String>
        get() = queue.poll(100, TimeUnit.MILLISECONDS)!!

    override fun run() {
        val map = createChildTtlInstancesAndModifyParentTtlInstances(tag, ttlInstances)
        queue.put(map)
    }
}
